staging-next 2024-10-15

vcunat opened the pull request:
This is the last iteration that can contain breaking changes to [release-critical packages](https://nixos.github.io/release-wiki/Release-Critical-Packages.html).

##### Helpful links

- workflow docs: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#staging
- important jobs:
  https://hydra.nixos.org/job/nixpkgs/staging-next/unstable#tabs-constituents
  https://hydra.nixos.org/job/nixos/staging-next-small/tested
- jobsets:
  https://hydra.nixos.org/jobset/nixpkgs/staging-next
  https://hydra.nixos.org/jobset/nixos/staging-next-small
- nix-review reports: https://malob.github.io/nix-review-tools-reports/
- previous staging-next: https://github.com/NixOS/nixpkgs/pull/343421
##### Important breakages
- [x] installer tests [more below](#issuecomment-2441081863)
- [x] virtualbox [more below](#issuecomment-2449688111)

emilazy committed:
openjdk: copy latest version into `generic.nix`

emilazy committed:
openjdk: format `generic.nix` with `nixfmt-rfc-style`

emilazy committed:
zulu: don’t depend on OpenJDK’s `meta.nix`

This is about to go away.

emilazy committed:
openjdk{8,11,17,21,23}: add update script

emilazy committed:
nixpkgs-openjdk-updater: init at 0.1.0

emilazy committed:
openjfx{17,21,23}: move to `pkgs/by-name`

reckenrode committed:
rcodesign: 0.22.0 -> 0.27.0

https://github.com/indygreg/apple-platform-rs/releases/tag/apple-codesign%2F0.23.0
https://github.com/indygreg/apple-platform-rs/releases/tag/apple-codesign%2F0.24.0
https://github.com/indygreg/apple-platform-rs/releases/tag/apple-codesign%2F0.25.0
https://github.com/indygreg/apple-platform-rs/releases/tag/apple-codesign%2F0.26.0
https://github.com/indygreg/apple-platform-rs/releases/tag/apple-codesign%2F0.27.0

reckenrode committed:
rcodesign: fix failing tests

- Ensure that uutils-coreutils is available for cli_tests;
- Fix logging (needed for cli_tests); and
- Disable tests that require network access.

reckenrode committed:
doc: update platform documentation for the new Darwin SDK pattern

reckenrode committed:
rcodesign: update to use the new SDK pattern on Darwin

yuanw committed:
haskellPackages: Enable darwin builds for some packages depending on mesa

update package.yaml

emilazy committed:
mpv: enable Vulkan on Darwin

Using `--vo=gpu` or `--vo=gpu-next` with MoltenVK is now supported
and has replaced the previous `cocoa-cb` as the default.

emilazy committed:
mpv: don’t override Lua with Lua

emilazy committed:
mpv: remove `swiftSupport` flag

mpv currently doesn’t even build successfully on macOS without Swift,
and there’s no use of Swift outside of the platform.

emilazy committed:
mpv: remove legacy Darwin SDK pattern detritus

Oh, this is satisfying.

Swift already propagates its corresponding SDK, so we don’t need
the patch.

emilazy committed:
openjfx: copy latest version into `generic.nix`

emilazy committed:
openjfx: format `generic.nix` with `nixfmt-rfc-style`

emilazy committed:
openjfx{17,21,23}: deduplicate into `generic.nix`

emilazy committed:
openjdk: remove existing update script

emilazy committed:
openjdk: remove unused patch

emilazy committed:
openjdk{8,11,17,21,23}: deduplicate into `generic.nix`

emilazy committed:
openjfx{17,21,23}: add update script

emilazy committed:
_experimental-update-script-combinators.sequence: handle silent scripts

Co-authored-by: Jan Tojnar <jtojnar@gmail.com>

drupol committed:
python312Packages.primp: init at 0.6.4

dotlambda committed:
Revert "mycli: disable checking to fix build"

This reverts commit f12c5fd4b8454954a8d03874b8b4d7214c9ca800.
The tests were broken because paramiko couldn't be imported, which was
fixed by 50d865aa192645e10064173422b394a12aecdfea.

dotlambda committed:
mycli: modernize

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into haskell-updates

dotlambda committed:
mycli: unpin sqlparse

github-actions[bot] committed:
Merge master into staging-next

sternenseemann committed:
Revert "haskell.packages.ghc910.extensions: 0.1.0.2 -> 0.1.0.3"

This reverts commit c3322f4af0e5e6a37acd8a0c70d18c7f5db6164a.

Reason for revert: extensions 0.1.0.3 doesn't support Cabal == 3.12.*.
Dependency conflicts due to extensions seem annoying to deal with.

sternenseemann committed:
haskell.packages.ghc9101: reflect hackage updates

github-actions[bot] committed:
Merge master into staging-next

JohnRTitor committed:
Revert "mycli: disable checking to fix build" (#349914)

skeuchel committed:
abseil-cpp_202407: apply pkgconfig fix for darwin

Artturin committed:
Merge branch 'master' into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

trofi committed:
python3Packages.dbt-adapters: fix the val

Without the change the eval fails on `staging-next` as:

    $ nix build --no-link -f. python3Packages.dbt-adapters
    error:
       … while evaluating a branch condition
         at pkgs/development/interpreters/python/passthrufun.nix:27:9:
           26|       func = name: value:
           27|         if lib.isDerivation value then
             |         ^
           28|           lib.extendDerivation (valid value || throw "${name} should use `buildPythonPackage` or `toPythonModule` if it is to be part of the Python packages set.") {} value

       … while evaluating a branch condition
         at /home/slyfox/dev/git/nixpkgs-staging-next/lib/customisation.nix:263:8:
          262|
          263|     in if missingArgs == {}
             |        ^
          264|        then makeOverridable f allArgs

       (stack trace truncated; use '--show-trace' to show the full, detailed trace)

       error: attribute 'pythonRelaxDeps' already defined at pkgs/development/python-modules/dbt-adapters/default.nix:30:3
       at pkgs/development/python-modules/dbt-adapters/default.nix:34:3:
           33|
           34|   pythonRelaxDeps = [ "protobuf" ];
             |   ^
           35|

K900 committed:
python3Packages.dbt-adapters: fix the val (#350074)

sternenseemann committed:
haskell.packages.ghc9101.{retrie,apply-refact}: rm broken overrides

Neither package support GHC 9.10 currently. Their corresponding plugins
are thus disabled in HLS.

sternenseemann committed:
Merge branch master into haskell-updates

niklaskorz committed:
zed-editor: change auto-update message

sternenseemann committed:
haskell.compiler.ghc98: 9.8.2 -> 9.8.3

https://www.haskell.org/ghc/download_ghc_9_8_3.html

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into haskell-updates

adamcstephens committed:
zed-editor: fix darwin (#329653)

Aleksanaa committed:
haskellPackages: Enable darwin builds for some packages depending on mesa (#349186)

github-actions[bot] committed:
Merge master into staging-next

vcunat committed:
python3Packages.protobuf: propagate pkgs.protobuf (#349270)

vcunat committed:
gnome: Minor Updates (#348874)

vcunat committed:
abseil-cpp_202407: apply pkgconfig fix for darwin (#350035)

github-actions[bot] committed:
Merge master into staging-next

vcunat committed:
Revert "bmake: 20240808 -> 20240921"

This reverts commit bf213776214687cf87f26212c1102ccccffbdb5b.

This wasn't a big rebuild on nixpkgs master, but now on staging-next
it would rebuild *-darwin stdenv, which would be too expensive
at this point.

github-actions[bot] committed:
Merge master into staging-next

emilazy committed:
moltenvk: adjust default log level

emilazy committed:
mpv: fix typo

emilazy committed:
mpv: remove obsolete Swift library path hack

This is now detected correctly as part of the SDK, as it should be.

emilazy committed:
mpv: remove `speexSupport` flag

Does nothing, removed upstream in 2012(!):
<https://github.com/mpv-player/mpv/commit/1fde09db6f4cee7347842261234082470dd3a2ca>.

emilazy committed:
mpv: remove `theoraSupport` flag

Does nothing, removed upstream in 2012(!):
<https://github.com/mpv-player/mpv/commit/1fde09db6f4cee7347842261234082470dd3a2ca>.

emilazy committed:
mpv: remove `libpngSupport` flag

Does nothing, removed upstream in 2012(!):
<https://github.com/mpv-player/mpv/commit/6a26b4a66504f701baf35e58467e55aea28c0ad5>.

emilazy committed:
mpv: remove `screenSaverSupport` flag

This is required as part of X11 support since the Meson transition.

emilazy committed:
mpv: remove `xineramaSupport` flag

Removed upstream:
<https://github.com/mpv-player/mpv/commit/200992f90c49fdc2d5605cf79ffaa5247fd62e3e>.

emilazy committed:
mpv: remove unused `libXxf86vm` dependency

emilazy committed:
mpv: remove unused `libGLU` dependency

sternenseemann committed:
haskell.compiler.ghc98: 9.8.3 -> 9.8.2

HLS can't be built yet with 9.8.3, so it doesn't make sense to make it a
default yet. Unfortunately we are still missing a ghc-lib-parser release
that works with ghc 9.8.3.

emilazy committed:
mpv: enable `dvdnavSupport` on non‐Linux

It builds fine on macOS.

emilazy committed:
mpv: disable `vaapiSupport` on Darwin

emilazy committed:
mpv: enable `{pipewire,pulse,wayland,x11}Support` on non‐Darwin

No reason for FreeBSD to miss out on the fun.

emilazy committed:
mpv: disable `sdl2Support` by default

Upstream’s Meson defaults, upstream’s CI binaries, and Arch
don’t enable this.

emilazy committed:
mpv: remove `xvSupport` flag

This prints out a big warning about how you shouldn’t be using
it. Upstream CI builds and Arch don’t enable it. There are other
backends of questionable worth that we don’t expose options for,
so let’s ditch this one.

sternenseemann committed:
haskellPackages.github-app-token: disable network dependent tests

sternenseemann committed:
haskellPackages: mark builds failing on hydra as broken

This commit has been generated by maintainers/scripts/haskell/mark-broken.sh based on
*evaluation [1809515](https://hydra.nixos.org/eval/1809515) of nixpkgs commit [e7735af](https://github.com/NixOS/nixpkgs/commits/e7735afbdce5f32dabb78e9393792625ff6d5eaa) as of 2024-10-21 19:40 UTC*
from the haskell-updates jobset on hydra under https://hydra.nixos.org/jobset/nixpkgs/haskell-updates

sternenseemann committed:
haskellPackages: update hackage and stackage (#346720)

emilazy committed:
mpv: remove redundant Meson flag

Since we’re doing auto‐detection anyway, this is unnecessary.

emilazy committed:
mpv: use wrapped bundle executable on Darwin

On macOS, many things won’t work properly unless mpv is executed from
the app bundle, such as spatial audio with `--ao=avfoundation`. This
ensures that those features work reliably and also avoids duplicating
the entire `mpv` binary.

fabianhjr committed:
Merge remote-tracking branch 'origin/master' into staging-next

github-actions[bot] committed:
Merge master into staging-next

K900 committed:
python312Packages.cfn-lint: 1.15.1 -> 1.18.1

Diff: https://github.com/aws-cloudformation/cfn-lint/compare/refs/tags/v1.15.1...v1.18.1

Changelog: https://github.com/aws-cloudformation/cfn-lint/blob/v1.18.1/CHANGELOG.md

drupol committed:
python312Packages.duckduckgo-search: 6.2.13 -> 6.3.2

drupol committed:
python312Packages.pyreqwest-impersonate: remove

The package has been replaced by `python312Packages.primp`.

sternenseemann committed:
haskellPackages.hledger*_1_40: wire up correctly

We forgot to update these overrides after 1.34 got removed from nixpkgs.
Let's get the latest versions, 1.40, to work.

fabianhjr committed:
Merge remote-tracking branch 'origin/master' into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

paparodeo committed:
python312Packages.typer: fix darwin build

checks require `ps` for darwin so add procps to nativeCheckInputs

reckenrode committed:
rcodesign: 0.22.0 -> 0.27.0 (#350374)

K900 committed:
python312Packages.typer: fix darwin build (#350586)

github-actions[bot] committed:
Merge master into staging-next

paparodeo committed:
lsof: fix darwin

edit Configure to use $SDKROOT to find libproc.h

paparodeo committed:
boost1{75,77,78,79}: fix build

condition a substituteInPlace only on boost 180 and above and add one
substituteInPlace back to the only consumer, boost175.

updates change 90df0c3ed8794b60e95bb105674dd3f6d7f81972

drupol committed:
python312Packages.primp: init at 0.6.4 (#349614)

github-actions[bot] committed:
Merge master into staging-next

K900 committed:
boost1{75,77,78,79}: fix build (#350658)

emilazy committed:
{open,}jdk{19,20,22}{,_headless}: don’t alias the `throw`s

This causes issues with the automated removal script.

github-actions[bot] committed:
Merge master into staging-next

paparodeo committed:
python312Packages.pytest-mpl: fix darwin

fix checkphase on darwin to not access the display by using a workaround
for matplotlib:
https://github.com/matplotlib/matplotlib/issues/26292#issuecomment-1632887774

mweinelt committed:
python312Packages.jupyter-sphinx: disable failing test

mweinelt committed:
python312Packages.cypari2: drop obsolete patch

mweinelt committed:
python312Packages.jaraco-functools: disable failing test on darwin

mweinelt committed:
python312Packages.proxy-py: fix tests on darwin

mweinelt committed:
python312Packages.pydantic-extra-types: disable failing test on darwin

paparodeo committed:
castxml: fix darwin build

clang on darwin adds an `-isysroot` flag which breaks the regular
expressions used in a couple tests

K900 committed:
Merge remote-tracking branch 'origin/master' into staging-next

paparodeo committed:
chibi: fix darwin

add libutil as required by link step

paparodeo committed:
vorbis-tools: fix darwin build

ld64 doesn't support linking with nested archives and as the nested
archived isn't required to build leave it out.

emilazy committed:
zed-editor: fix incorrect manual merge

emilazy committed:
zed-editor: fix incorrect manual merge (#350896)

vcunat committed:
python3Packages.pyre-extensions: fixup src download

https://hydra.nixos.org/build/276236752/nixlog/11/tail
No idea what's going on here.  One-time typo upstream?

vcunat committed:
chibi: fix darwin (#350883)

emilazy committed:
vorbis-tools: fix darwin build (#350890)

fpletz committed:
mpv: clean‐ups and Darwin improvements (#350674)

K900 committed:
kanidm: fix build with Rust 1.82

mweinelt committed:
python312Packages.click-odoo-contrib: fix src pname

github-actions[bot] committed:
Merge master into staging-next

mweinelt committed:
paperless-ngx: disable failing tests

These are due to a test-only dependency, so no functionality should be
impaired.

emilazy committed:
python312Packages.pytest-mpl: fix darwin (#350784)

thiagokokada committed:
open{jdk{8,11,17,21,23},jfx{17,21,23}}: deduplicate (#350736)

github-actions[bot] committed:
Merge master into staging-next

emilazy committed:
lsof: fix darwin (#350651)

emilazy committed:
castxml: fix darwin build (#350853)

paparodeo committed:
gtk4: fix x64 darwin

use 10.15 sdk to fix build and propagate the sdk and min sdk version (10.15)

emilazy committed:
gtk4: fix x64 darwin (#351004)

drupol committed:
open-webui: 0.3.32 -> 0.3.35

diff: https://github.com/open-webui/open-webui/compare/v0.3.32..v0.3.35
changelog: https://github.com/open-webui/open-webui/releases/tag/v0.3.35

emilazy committed:
mpv: remove unreferenced patch

Fixes: f2280510e3e431e9c0907fe233a3534def4a7162

emilazy committed:
mpv: remove unreferenced patch (#351026)

mweinelt committed:
python312Packages.python-snap7: convert to pep517 build

paparodeo committed:
python312Packages.pycurl: fix darwin build

don't link with flat_namespace on darwin as it results in the runtime
linker/loader pulling in the system libcurl rather than nixpkgs one.

https://github.com/pycurl/pycurl//commit/7deb85e24981e23258ea411dcc79ca9b527a297d

emilazy committed:
python312Packages.pycurl: fix darwin build (#351043)

fabianhjr committed:
Merge remote-tracking branch 'origin/master' into staging-next

paparodeo committed:
gtk4:  fix darwinMinVersionHook typo

set min version to 10.15 not 15.0

emilazy committed:
gtk4:  fix darwinMinVersionHook typo (#351053)

K900 committed:
Merge remote-tracking branch 'origin/master' into staging-next

paparodeo committed:
lldb_{14,15,16,17,18,19}: fix x64 darwin

x64 darwin doesn't build with 10.12 sdk after the sdk refactor due to
compile errors. bumping the sdk to 11 fixes the build.

set min deployment target as needed for lldb_14

lldb_14 still broken but due to an unrelated issue.

fabianhjr committed:
obs-studio: add `-Wno-error=deprecated-declarations` due to build failure

K900 committed:
obs-studio: add `-Wno-error=deprecated-declarations` due to build failure (#351075)

github-actions[bot] committed:
Merge master into staging-next

drupol committed:
python312Packages.rtfde: fix build

I updated the `disabledTests` and added `lark` in `pythonRelaxDeps`.

drupol committed:
python311Packages.xhtml2pdf: fix build

- Remove the patch, it was already included in 0.2.16.
- Remove `pythonRelaxDeps`, it's not needed anymore

skeuchel committed:
openjdk: add riscv64-linux to platforms

paparodeo committed:
stockfish: fix x64 darwin

use sdk 11 and set min version to 10.15 to match the makefiles

paparodeo committed:
samba: fix darwin

the cups headers are part of the sdk but the library is not, which
causes configure to detect cups but the build fails with a link error.
explicitly enable / disable cups based on `enablePrinting` flag.

emilazy committed:
openjdk: add riscv64-linux to platforms (#343907)

emilazy committed:
javaPackages.openjdk{8,11,17}-bootstrap: fix eval on Darwin

K900 committed:
Merge remote-tracking branch 'origin/master' into staging-next

github-actions[bot] committed:
Merge master into staging-next

emilazy committed:
lldb_{14,15,16,17,18,19}: fix x64 darwin (#351084)

emilazy committed:
stockfish: fix x64 darwin (#351114)

emilazy committed:
doc: update platform documentation for the new Darwin SDK pattern (#350395)

drupol committed:
python311Packages.xhtml2pdf: fix build (#351113)

drupol committed:
python312Packages.rtfde: fix build (#351110)

jackwilsdon committed:
python3Packages.passlib: fix passlib.pwd

paparodeo committed:
qrupdate: fix darwin

with the sdk refactor cmake will use Accelerate for the blas/lapack
libs. Unfortunately, this breaks qrupdate and causes all test to fail,
most of them by segfaulting. Set the Blas vendor to Generic to prevent
cmake from using Accelerate and use the nixpkgs provided blas/lapack
libraries

paparodeo committed:
rehex: fix darwin

add libiconv to fix darwin build and remove now unneeded frameworks

paparodeo committed:
python312Packages.watchdog: fix x64 darwin

drop sdk 10.12 patch that no longer applies and use sdk 11

vcunat committed:
samba: fix darwin (#351117)

github-actions[bot] committed:
Merge master into staging-next

paparodeo committed:
libticonv: link with libiconv

the darwin sdk rework removed the libiconv hook which adds -liconv so
add explicitly to work around broken autoconf libiconv detection

paparodeo committed:
rsstail: 2.1 -> 2.2

fix linking to libiconv on darwin

https://github.com/folkertvanheusden/rsstail/compare/6f2436185372b3f945a4989406c4b6a934fe8a95...v2.2

emilazy committed:
qrupdate: fix darwin (#351337)

github-actions[bot] committed:
Merge master into staging-next

emilazy committed:
python312Packages.watchdog: fix x64 darwin (#351344)

emilazy committed:
rehex: fix darwin (#351339)

paparodeo committed:
ripmime: link to -liconv for darwin

emilazy committed:
libticonv: link with libiconv (#351407)

emilazy committed:
ripmime: link to -liconv for darwin (#351425)

JohnRTitor committed:
opencl-headers: 2023.12.14 -> 2024.05.08 (#310255)

collares committed:
sage: import sympy and numpy update patches

mweinelt committed:
python312Packages.pyvirtualdisplay: modernize, enable tests on linux

mweinelt committed:
python312Packages.mss: modernize, test on linux

K900 committed:
Merge remote-tracking branch 'origin/master' into staging-next

paparodeo committed:
stfl: link with libiconv on darwin

paparodeo committed:
rsstail: adjust license to gpl2Only

emilazy committed:
rsstail: 2.1 -> 2.2 and libiconv darwin fix (#351414)

paparodeo committed:
stfl: 0.24 -> 0.24-unstable-2021-11-29

update repo to https://github.com/newsboat/stfl as previous is no longer
available

paparodeo committed:
stfl: update homepage

emilazy committed:
stfl: link with libiconv on darwin (#351499)

mweinelt committed:
python312Packages.{mss,pyvirtualdisplay}: modernize, enable tests on linux (#351482)

github-actions[bot] committed:
Merge master into staging-next

reckenrode committed:
ispc: drop impure Xcode dependency

The new Darwin SDK pattern allows ispc to be built without requiring an
impure Xcode.

reckenrode committed:
ispc: drop impure Xcode dependency (#351544)

zowoq committed:
gitstatus: link with libiconv on darwin

https://hydra.nixos.org/build/276133321/nixlog/2

K900 committed:
javaPackages.openjdk{8,11,17}-bootstrap: fix eval on Darwin (#351160)

K900 committed:
Merge remote-tracking branch 'origin/master' into staging-next

collares committed:
[staging-next] sage: import sympy and numpy update patches (#351470)

SuperSandro2000 committed:
gitstatus: link with libiconv on darwin (#351546)

GaetanLepage committed:
mission-center: 0.6.1 -> 0.6.2

Diff: https://gitlab.com/mission-center-devs/mission-center/-/compare/v0.6.1...v0.6.2

Changelog: https://gitlab.com/mission-center-devs/mission-center/-/releases/v0.6.2

ofalvai committed:
psst: unstable-2024-10-07 -> unstable-2024-10-24

ofalvai committed:
otel-desktop-viewer: fix darwin build

paparodeo committed:
cdrkit: darwin link with CF

fabianhjr committed:
Merge remote-tracking branch 'origin/master' into staging-next

github-actions[bot] committed:
Merge master into staging-next

dotlambda committed:
python3Packages.passlib: fix passlib.pwd (#351332)

fabianhjr committed:
Merge remote-tracking branch 'origin/master' into staging-next

github-actions[bot] committed:
Merge master into staging-next

paparodeo committed:
bb: fix darwin / clang

- autoreconfHook
- add missing headers needed for function prototypes.
- remove regparm for clang

vcunat committed:
Revert "python3Packages.pymilvus: 2.3.6 -> 2.4.7"

This reverts commit 6fe5dbd9503caf88f39c6a786b1bb86522c257a2.
Fixes #351102
(for now; feel free to upgrade and fix)

paparodeo committed:
ashuffle: link to CF on darwin

vcunat commented:
- [ ] installer tests have insufficient dependencies again, e.g. `nixosTests.installer.simpleUefiSystemdBoot` :
  https://hydra.nixos.org/build/276424361

vcunat committed:
azure-cli: 2.64.0 -> 2.65.0 (#347241)

K900 committed:
Merge remote-tracking branch 'origin/master' into staging-next

emilazy committed:
tests.build-environment-info: add more information

The user information would have helped diagnose the empty GECOS issue
we had with the Hydra Darwin builders recently and the OS information
seems generally nice to have.

emilazy committed:
release-lib: build packages without `meta.platforms` on all platforms

This was added in <https://github.com/NixOS/nixpkgs/pull/19990> to
ensure that packages that previously weren’t building at all would
get at least one platform’s worth of coverage. However, since most
`runCommand` calls don’t explicitly set any `meta` information,
this meant that things like `tests.makeWrapper` were only being
run for `x86_64-linux`. Since most trivial builder calls should
work on all platforms, it doesn’t make much sense for them to be
restricted in this way, and we shouldn’t unnecessarily penalize
other platforms by treating an empty `meta.platforms` inconsistently
with `check-meta`’s maximally‐permissive interpretation. Actual
packages that only work on a subset of platforms should, of course,
set `meta.platforms` explicitly.

drupol committed:
python312Packages.langchain-community: relax `tenacity` dependency

emilazy committed:
ashuffle: link to CF on darwin (#351821)

emilazy committed:
cdrkit: darwin link with CF (#351761)

mweinelt committed:
python312Packages.tpm2-pytss: drop obsolete patch

K900 committed:
Merge remote-tracking branch 'origin/master' into staging-next

drupol committed:
python312Packages.langchain-community: relax `tenacity` dependency (#351941)

drupol committed:
python311.fake-useragent: fix build

drupol committed:
python311.fake-useragent: fix build (#351959)

emilazy committed:
cudaPackages.cuda-samples: set `meta.platforms`

emilazy committed:
cudaPackages.cuda-library-samples: set `meta.platforms`

emilazy committed:
phpExtensions.*: use `meta.broken` rather than filtering

github-actions[bot] committed:
Merge master into staging-next

emilazy committed:
phpExtensions.*: inherit `meta.platforms` from PHP

emilazy committed:
tests.nixos-functions: don’t omit derivations on non‐Linux

Hydra can handle packages that `throw` but not attributes that
disappear depending on platform, and it seems okay for evaluation to
fail here for non‐Linux platforms.

github-actions[bot] committed:
Merge master into staging-next

K900 committed:
Merge remote-tracking branch 'origin/master' into staging-next

github-actions[bot] committed:
Merge master into staging-next

Aleksanaa committed:
otel-desktop-viewer: fix darwin build (#351732)

tjni committed:
python312Packages.bilibili-api-python: fix 16.3.0 build

The published tar.gz name changed from using hyphens to underscores
(likely because of change in setuptools 72.2.0).

At the same time, I noticed that the CHANGELOG removes requests as a
dependency, so I've done the same here.

github-actions[bot] committed:
Merge master into staging-next

emilazy committed:
texlivePackages.*: set `meta.{platforms,hydraPlatforms}`

JohnRTitor committed:
mission-center: 0.6.1 -> 0.6.2 (#351653)

tjni committed:
python312Packages.bilibili-api-python: fix 16.3.0 build (#352116)

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

drupol committed:
open-webui: 0.3.32 -> 0.3.35 (#351022)

github-actions[bot] committed:
Merge master into staging-next

RossComputerGuy committed:
release-lib: build packages without `meta.platforms` on all platforms (#351919)

vcunat committed:
psst: unstable-2024-10-07 -> unstable-2024-10-24 (#351730)

https://hydra.nixos.org/build/276403899/nixlog/17/tail

K900 committed:
Merge remote-tracking branch 'origin/master' into staging-next

paparodeo committed:
xorg.libAppleWM: 1.4.1 -> 1.4.1-unstable-2021-01-04

fixes build on darwin

https://gitlab.freedesktop.org/xorg/lib/libapplewm/-/compare/libAppleWM-1.4.1...d8926ba5805ac8feed0b0cf8ebf54d8ab7af2314

emilazy committed:
xorg.libAppleWM: 1.4.1 -> 1.4.1-unstable-2021-01-04 (#351776)

rnhmjoj committed:
bb: fix darwin / clang (#351820)

Pandapip1 commented:
According to https://github.com/NixOS/nixpkgs/issues/339153, this PR is to be merged today. When should I expect this to be merged?

emilazy committed:
auto-patchelf: avoid filesets

The use of filesets triggers the same Nix evaluation bug as in
e4b8a0a243d917d78935e6c479a5005e7c52063f, which broke the installer
tests again.

emilazy commented:
It’s `staging`. Time is elastic.

emilazy commented:
(But not before https://github.com/NixOS/nixpkgs/pull/352491, at least.)

mweinelt committed:
python312Packages.pycurl: disable crashing test

emilazy committed:
auto-patchelf: avoid filesets (#352491)

github-actions[bot] committed:
Merge master into staging-next

yuyuyureka committed:
tree-wide: switch initrd generators back to gnu cpio

Originally, we switched to bsdtar from libarchive to solve a reproducibility issue related to hardlinks

As of gnu cpio 2.14 the --ignore-dirnlink option is introduced and now included in --reproducible, which solves this issue

By switching back, we are in turn solving an issue in libarchive >=3.7.5 erroring out with "Error reading archive -: (null)"

Change-Id: Ib6140d599b6547d8e941b0251ce996e303c41fa6

emilazy committed:
auto-patchelf: rename source directory to avoid rebuilds

emilazy committed:
auto-patchelf: rename source directory to avoid rebuilds (#352509)

paparodeo committed:
cconv: link -liconv on darwin

mweinelt committed:
python313Packages.botocore-stubs: fix build-system

mweinelt committed:
esphome: relax wheel constraint

emilazy committed:
[staging-next] tree-wide: switch initrd generators back to gnu cpio (#352498)

emilazy committed:
cconv: link -liconv on darwin (#352512)

RossComputerGuy commented:
> When should I expect this to be merged?

I'm the release manager so I think I'll be the one doing it. Hopefully sometime soon ™️ .

emilazy commented:
Probably we should let @vcunat handle it as usual? We haven’t even gotten a `staging-next-small` eval started yet.

RossComputerGuy commented:
Probably best.

github-actions[bot] committed:
Merge master into staging-next

paparodeo committed:
Revert "bisq2: init @ 2.1.2"

This reverts commit 7367fb0f47cbb9c324bebf973596838eed965fd1.

commit breaks eval due as it uses openjdk 22 which was removed in
staging-next

vcunat committed:
Revert "bisq2: init @ 2.1.2" (#352550)

vcunat commented:
I believe you've fixed all known blockers, and I looked at staging-next-small to get more confidence.  So I see no need to delay anymore.

vcunat commented:
- [ ] `virtualbox*` are broken, blocking the nixos-unstable channel, I'm afraid.
  https://hydra.nixos.org/build/276715853
  maintainers: @svanderburg, @friedrichaltheide, @blitz

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/systemd-boot-failing-to-boot-a-nixos-warbler-system-due-to-an-efi-assertion-failure/61450/9
